fabric-rendering-v1/src/client/java/net/fabricmc/fabric/impl/client/rendering/hud/HudStatusBarHeightRegistryImpl.java

```diff
@@ -0,0 +1,394 @@
+		int rowShift = Math.max(10 - (healthRows - 2), 3);
+		return 10 + (healthRows - 1) * rowShift;
+	};
+	/**
+	 * Height provider for the vanilla armor bar.
+	 *
+	 * <p>Mods that would otherwise have a mixin for this are encouraged to instead register a replacement provider via
+	 * {@link net.fabricmc.fabric.api.client.rendering.v1.hud.HudStatusBarHeightRegistry#addLeft(Identifier,
+	 * StatusBarHeightProvider)}.
+	 */
+	static final StatusBarHeightProvider ARMOR_BAR = (PlayerEntity player) -> {
+		return player.getArmor() > 0 ? 10 : 0;
+	};
+	/**
+	 * Height provider for the vanilla mount health.
+	 *
+	 * <p>Mods that would otherwise have a mixin for this are encouraged to instead register a replacement provider via
+	 * {@link net.fabricmc.fabric.api.client.rendering.v1.hud.HudStatusBarHeightRegistry#addRight(Identifier,
+	 * StatusBarHeightProvider)}.
+	 */
+	static final StatusBarHeightProvider MOUNT_HEALTH = (PlayerEntity player) -> {
+		InGameHud hud = MinecraftClient.getInstance().inGameHud;
+		LivingEntity livingEntity = ((InGameHudAccessor) hud).callGetRiddenEntity();
+		int vehicleMaxHearts = ((InGameHudAccessor) hud).callGetHeartCount(livingEntity);
+		return ((InGameHudAccessor) hud).callGetHeartRows(vehicleMaxHearts) * 10;
+	};
+	/**
+	 * Height provider for the vanilla food bar.
+	 *
+	 * <p>Mods that would otherwise have a mixin for this are encouraged to instead register a replacement provider via
+	 * {@link net.fabricmc.fabric.api.client.rendering.v1.hud.HudStatusBarHeightRegistry#addRight(Identifier,
+	 * StatusBarHeightProvider)}.
+	 */
+	static final StatusBarHeightProvider FOOD_BAR = (PlayerEntity player) -> {
+		InGameHud hud = MinecraftClient.getInstance().inGameHud;
+		LivingEntity livingEntity = ((InGameHudAccessor) hud).callGetRiddenEntity();
+		return ((InGameHudAccessor) hud).callGetHeartCount(livingEntity) == 0 ? 10 : 0;
+	};
+	/**
+	 * Height provider for the vanilla air bar.
+	 *
+	 * <p>Mods that would otherwise have a mixin for this are encouraged to instead register a replacement provider via
+	 * {@link net.fabricmc.fabric.api.client.rendering.v1.hud.HudStatusBarHeightRegistry#addRight(Identifier,
+	 * StatusBarHeightProvider)}.
+	 */
+	static final StatusBarHeightProvider AIR_BAR = (PlayerEntity player) -> {
+		int maxAirSupply = player.getMaxAir();
+		int airSupply = Math.clamp(player.getAir(), 0, maxAirSupply);
+		boolean isInWater = player.isSubmergedIn(FluidTags.WATER);
+		return isInWater || airSupply < maxAirSupply ? 10 : 0;
+	};
+	/**
+	 * This serves two purposes: it provides a fixed order for some vanilla status bars; and it provides reduced vanilla
+	 * height providers, to compare with the actual height providers during rendering for potential translations for
+	 * vanilla status bars. Translations are achieved via matrix stack transformations; alternatively can also be
+	 * implemented via mixins.
+	 *
+	 * <p>Do not use {@link Map}, it does not preserve insertion order.
```

Comment:
But the field is a `Map`?

Reply:
Specifically don't use `Map#of`, as it has unspecified iteration order.